cmd/minishift/cmd/console.go

```diff
@@ -45,9 +46,19 @@ var consoleCmd = &cobra.Command{
 	Short:   "Opens or displays the OpenShift Web Console URL.",
 	Long:    `Opens the OpenShift Web Console URL in the default browser or displays it to the console.`,
 	Run: func(cmd *cobra.Command, args []string) {
+
 		api := libmachine.NewClient(state.InstanceDirs.Home, state.InstanceDirs.Certs)
 		defer api.Close()
 
+		util.ExitIfUndefined(api, constants.MachineName)
+
+		host, err := api.Load(constants.MachineName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		util.ExitIfNotRunning(host.Driver, constants.MachineName)
+
```

Comment:
@bhavin192 I think it would be better to improve the getHostUrl function as it is called by `console` and all the flags. I have created a code to test this at https://github.com/LalatenduMohanty/minishift/commit/e85a295efe9e89092d656b23171a8403f51d1dc7 .